cdap-app-templates/cdap-etl/hydrator-spark-core/src/main/java/co/cask/cdap/etl/spark/batch/DatasetInfo.java

```diff
@@ -41,12 +33,6 @@
   private final Map<String, String> datasetArgs;
   private final List<Split> splits;
 
-  public static DatasetInfo deserialize(DataInput input) throws IOException {
-    return new DatasetInfo(input.readUTF(),
-                           Serializations.deserializeMap(input, Serializations.createStringObjectReader()),
-                           deserializeSplits(input));
-  }
-
   DatasetInfo(String datasetName, Map<String, String> datasetArgs, @Nullable List<Split> splits) {
```

Comment:
Looks like a bug to me. If the splits are specified, we should use them. I have added a third commit to fix it.